### PR TITLE
Return default start value according to type

### PR DESCRIPTION
**Pull request by lochel**

### CODE REVIEW

File: OMCompiler/Compiler/FrontEnd/DAEUtil.mo [hunk 1]
```diff
@@ -909,9 +909,13 @@ algorithm
...
       DAE.Exp r;
     case (SOME(DAE.VAR_ATTR_REAL(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_REAL(start = NONE()))) then DAE.RCONST(0.0);
     case (SOME(DAE.VAR_ATTR_INT(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_INT(start = NONE()))) then DAE.ICONST(0);
     case (SOME(DAE.VAR_ATTR_BOOL(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_BOOL(start = NONE()))) then DAE.BCONST(false);
     case (SOME(DAE.VAR_ATTR_STRING(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_STRING(start = NONE()))) then DAE.SCONST("");
     case (SOME(DAE.VAR_ATTR_ENUMERATION(start = SOME(r)))) then r;
```
Comment (lochel) [inline OMCompiler/Compiler/FrontEnd/DAEUtil.mo:919]: What is the default value for enumerations?

Comment (perost) [reply OMCompiler/Compiler/FrontEnd/DAEUtil.mo:919]: The first enumeration value according to [the specification](https://specification.modelica.org/master/class-predefined-types-and-declarations.html#predefined-types-and-classes):
```Modelica
parameter EnumType start = e1;       // Initial value
```

**Comment by lochel** [comment]
We are still not handling enumerations though :-/ Anyone volunteering to fix it?

**Comment by hkiel** [comment]
I cannot compile (macOS) right now, but I guess the following would be fine?
```
diff --git a/OMCompiler/Compiler/FrontEnd/DAEUtil.mo b/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
index 68822f4193..02a5e0f56f 100644
--- a/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
+++ b/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
@@ -908,6 +908,7 @@ algorithm
   start := match(inVariableAttributesOption)
     local
       DAE.Exp r;
+      Absy.Path path;
     case (SOME(DAE.VAR_ATTR_REAL(start = SOME(r)))) then r;
     case (SOME(DAE.VAR_ATTR_REAL(start = NONE()))) then DAE.RCONST(0.0);
     case (SOME(DAE.VAR_ATTR_INT(start = SOME(r)))) then r;
@@ -917,6 +918,7 @@ algorithm
     case (SOME(DAE.VAR_ATTR_STRING(start = SOME(r)))) then r;
     case (SOME(DAE.VAR_ATTR_STRING(start = NONE()))) then DAE.SCONST("");
     case (SOME(DAE.VAR_ATTR_ENUMERATION(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_ENUMERATION(path = path, start = NONE()))) then DAE.ENUM_LITERAL(path=path, index=1);
     else DAE.RCONST(0.0);
   end match;
 end getStartAttr;
```

**Comment by hkiel** [comment]
Would this be enough, or do we also need the `Absyn.Path path` in `DAE.ENUM_LITERAL`?
```
diff --git a/OMCompiler/Compiler/FrontEnd/DAEUtil.mo b/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
index 68822f4193..4843482b86 100644
--- a/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
+++ b/OMCompiler/Compiler/FrontEnd/DAEUtil.mo
@@ -917,6 +917,7 @@ algorithm
     case (SOME(DAE.VAR_ATTR_STRING(start = SOME(r)))) then r;
     case (SOME(DAE.VAR_ATTR_STRING(start = NONE()))) then DAE.SCONST("");
     case (SOME(DAE.VAR_ATTR_ENUMERATION(start = SOME(r)))) then r;
+    case (SOME(DAE.VAR_ATTR_ENUMERATION(start = NONE()))) then DAE.ENUM_LITERAL(index=1);
     else DAE.RCONST(0.0);
   end match;
 end getStartAttr;
```

**Pull request merged**